### DISCUSSION TITLE
Question about integrating cuRobo with a Real UR Robot in a ROS2 Environment

### DISCUSSION THREAD

**Discussion by angeelalg** [NVlabs/curobo#187] [Software Q&A] [answered]
Hello!

I'm impressed with the speed results of cuRobo, compared to the classic OMPL planners available on Moveit! which I'm very used to.

I would like to be able to use cuRobo in some real UR arm integrated with my whole system (already existing in ROS2). How would be the setup to be able to command a real UR robot with cuRobo from ROS2? is it even something feasible?
I don't have any experience with Isaac Sim...

Thank you very much in advance.

**Comment by balakumar-s** [comment, marked as answer]
You don't need Isaac Sim to run cuRobo on real UR. You can use UR's scaled_joint_trajectory_controller to execute trajectories from cuRobo. A python implementation of sending joint trajectories is here: https://github.com/fmauch/Universal_Robots_ROS2_Driver/blob/test_move_script/ur_robot_driver/scripts/test_move.py. In that code, swap out the trajectories with trajectories from cuRobo.
